test/Staking.test.js

```diff
@@ -0,0 +1,299 @@
+      it("Stake by acc2 false - Error : staking has not started yet", async () => {
+        let balanceBefore = await instanceToken.balanceOf(acc2);
+        expect(balanceBefore).to.be.bignumber.equal(ether('100'));
+        await instanceToken.approve(instanceStaking.address, ether('100'), { from: acc2 });
+        await expectRevert(instanceStaking.stake(ether('60'), { from: acc2 }), "Error : staking has not started yet");
+        let balanceAfter = await instanceToken.balanceOf(acc2);
+        expect(balanceAfter).to.be.bignumber.equal(ether('100'));
+      });
+    });
+
+    describe("Done", async () => {
+      it("SetRewards - done function", async () => {
+        let balanceBeforeOwner = await instanceToken.balanceOf(owner);
+        expect(balanceBeforeOwner).to.be.bignumber.equal(ether('5600000'));
+        let balanceBeforeContract = await instanceToken.balanceOf(instanceStaking.address);
+        expect(balanceBeforeContract).to.be.bignumber.equal(ether('0'));
+        let start = await time.latest();
+        await instanceToken.approve(instanceStaking.address, ether('500000'));
+        let tx = await instanceStaking.setRewards(start.add(new BN(60)), ether('500000'), ether('10'));
+        expectEvent(tx, "SetRewards", { startTime: start.add(new BN(60)), reward: ether('500000'), apr: ether('10') });
+        let balanceAfterOwner = await instanceToken.balanceOf(owner);
+        expect(balanceAfterOwner).to.be.bignumber.equal(ether('5100000'));
+        let balanceAfterContract = await instanceToken.balanceOf(instanceStaking.address);
+        expect(balanceAfterContract).to.be.bignumber.equal(ether('500000'));
+      });
+    });
+  });
+
+  describe("Stake", async () => {
+
+    describe("False", async () => {
+      it("Stake by acc2 false - Error : you can't stake 0 tokens", async () => {
+        let balanceBefore = await instanceToken.balanceOf(acc2);
+        expect(balanceBefore).to.be.bignumber.equal(ether('100'));
+        await instanceToken.approve(instanceStaking.address, ether('100'), { from: acc2 });
+        await expectRevert(instanceStaking.stake(ether('0'), { from: acc2 }), "Error : you can't stake 0 tokens");
+        let balanceAfter = await instanceToken.balanceOf(acc2);
+        expect(balanceAfter).to.be.bignumber.equal(ether('100'));
+      });
+      it("Stake by acc2 false - Error : your stake value too high", async () => {
+        let balanceBefore = await instanceToken.balanceOf(owner);
+        expect(balanceBefore).to.be.bignumber.equal(ether('5100000'));
+        await instanceToken.approve(instanceStaking.address, ether('5000001'));
+        await expectRevert(instanceStaking.stake(ether('5000001'), { from: owner }), "Error : your stake value too high");
+        let balanceAfter = await instanceToken.balanceOf(owner);
+        expect(balanceAfter).to.be.bignumber.equal(ether('5100000'));
+      });
+    });
+
+    describe("Done", async () => {
+      it("Stake by acc2 - done", async () => {
```

Comment:
very bad name for tests, should change to more corectly such as:
"Success stake from second user on 100 tokens" etc